### ISSUE TITLE
v0.8.0 release planning

### ISSUE THREAD

**Issue by davidanthoff** [JuliaEditorSupport/julia-vscode#247]
I assume we add [DocumentFormat.jl](https://github.com/ZacLN/DocumentFormat.jl) as a dependency, right? Should I put the necessary submodule into place already? Can't hurt, right?

**Comment by ZacLN** [comment]
Yep add it in

**Comment by ZacLN** [comment]
I'm not _completely_ sure we need a separate process for parsing anymore. Kristoffer and I have done a  fair bit of work on CSTParser and sped it up by >6x (since LS v0.7.0 / CSTParser 0.1.2).

**Comment by davidanthoff** [comment]
Yeah, lets leave the parsing int he same process with all the amazing work you guys did on the parsing speed. Plus, we can move it into a background thread in the same process once that is supported.

So I think we really just would like to have a separate process for the loading of the user modules.

**Comment by ZacLN** [comment]
Should we set things up for a minor release? There's so many stability/speed improvements in there ready to go

**Comment by KristofferC** [comment]
Are you planning on retagging CSTParser 0.3 with the new PR?

**Comment by ZacLN** [comment]
yeh, I'll do that now

**Comment by davidanthoff** [comment]
Yes, lets do a release! I think we can just call it v0.8.0, after all there are some new features.

Are the following the right package versions to use:
- CSTParser v0.3.0
- DocumentFormat v0.1.0
- JSON v0.14.0
- Tokenize v0.4.1

We probably still need a new tag for LanguageServer, right? Do we need to merge https://github.com/JuliaEditorSupport/LanguageServer.jl/pull/183 first?

**Comment by ZacLN** [comment]
Yup, we should ship w/ Tokenize 0.4.1 but only require v0.4.0

**Comment by davidanthoff** [comment]
@ZacLN I just went through https://github.com/JuliaEditorSupport/LanguageServer.jl/milestone/2 and https://github.com/JuliaEditorSupport/julia-vscode/milestone/8 to figure out what we still need to do for a v0.8.0 release, but for most issues I wasn't sure :) Could you maybe close those that are solved, move anything we don't want to do in this cycle to the backlog? And then we can tag a v0.2.0 for LS, and a beta for the extension, I'd say?

**Comment by ZacLN** [comment]
Ok I've closed/backlogged as appropriate so we should be good to go

**Comment by davidanthoff** [comment]
https://github.com/JuliaEditorSupport/julia-vscode/releases/tag/v0.8.0-beta.1

Next step would probably be to do the beta announcement on the forum, right? I'm a bit swamped with stuff, so would probably be a couple of days for me to get to that. @ZacLN maybe you want to make that announcement this time? You probably also have a better sense of what is new :)

**Comment by ZacLN** [comment]
Yep I'll do it tomorrow evening, and try to get all the features mentioned this time :)

**Comment by davidanthoff** [comment]
Just a heads up, I'll be offline until Tue.

**Comment by ZacLN** [comment]
Coolio

**Comment by davidanthoff** [comment]
Should we shoot for a beta 2 release tomorrow, Fri? @ZacLN I guess you need to tag a CSTParser.jl release for that, then I'll tag a LanguageServer.jl release, and then I'll create the VSIX tomorrow?

**Comment by ZacLN** [comment]
Yep, I"ll tag it now

**Comment by ZacLN** [comment]
Hold up, there's another crash causing lint error I need to fix, will do so tonight

**Comment by davidanthoff** [comment]
https://github.com/JuliaEditorSupport/julia-vscode/releases/tag/v0.8.0-beta.2

**Comment by ZacLN** [comment]
We missed the julia-vscode pr in the new beta, should probably release a new beta (not necessarily now) before the release proper

**Comment by davidanthoff** [comment]
Yep, I just realized that. Sorry, should have been more careful. I'll package a new beta right away, doesn't make sense that folks test something that is out-of-date already.

**Comment by davidanthoff** [comment]
Now with the REPL variable explorer:

https://github.com/JuliaEditorSupport/julia-vscode/releases/tag/v0.8.0-beta.4

**Comment by davidanthoff** [comment]
Alright, I'm somewhat back. Still a lot of other things on my plate, but I should be able to do beta releases/final releases this week.

I do think we need another beta, right? My sense it the following things need to happen for that:
- [ ] CSTParser needs to tag version v0.3.3
- [ ] On the LS side, we need to merge https://github.com/JuliaEditorSupport/LanguageServer.jl/pull/211 and https://github.com/JuliaEditorSupport/LanguageServer.jl/pull/212, and then we'll tag v0.2.2.
- [ ] In the extension we'll need to merge https://github.com/JuliaEditorSupport/julia-vscode/pull/295 and https://github.com/JuliaEditorSupport/julia-vscode/pull/305.
- [ ] We need to sort out which of the things on the v0.8.0 milestone we still want to fix, move to a later release or have fixed already. Right now we still have 12 open issues, before we release a beta we should try to make that list much smaller.

@ZacLN anything else?

**Comment by ZacLN** [comment]
Do we need releases for a beta or can we just update the submodules? I've experience the plot pane [freezing](https://github.com/JuliaEditorSupport/julia-vscode/issues/314#issuecomment-338740733) which I'd like to fix. I'll have a look through the v0.8.0 issues in the next day or two

**Comment by davidanthoff** [comment]
Alright, I think we are down to three blocking things before we can release the next beta:
- https://github.com/JuliaEditorSupport/julia-vscode/issues/343 (I feel we could move this to v0.9.0 if too complicated)
- https://github.com/JuliaEditorSupport/julia-vscode/issues/342 (same here, if this takes more time we could move this to v0.9.0 and for now just restore the old key bindings)
- https://github.com/JuliaEditorSupport/LanguageServer.jl/pull/216 (this one we need, I believe, just waiting for @ZacLN review)

@ZacLN anything else? I think it would be best if you make a decision about the first two, you understand that part of the code best.

**Comment by ZacLN** [comment]
All clear on those issues. Thoughts on changing the `evalBlock` so when sat inside a module declaration it runs the inner block not the whole module?

**Comment by davidanthoff** [comment]
Yeah, I think it should run the inner block, ideally eval into the module it is in, right?

**Comment by davidanthoff** [comment]
But should we leave that to v0.9.0? I think we should get v0.8.0 out, too much good stuff in there to keep it from folks :)

**Comment by ZacLN** [comment]
Alrighty! One problem I've noticed that you may be able to help with as I think you faced the same thing with the plot pane: when you eval code from an editor in column 2, it reopens it in column 1...

**Comment by davidanthoff** [comment]
I think #350 should fix that.

Once that is merged: should we release a release candidate instead of a beta? I kind of feel we are ready for that phase, right?

**Comment by davidanthoff** [comment]
https://github.com/JuliaEditorSupport/julia-vscode/releases/tag/v0.8.0-rc.1

**Comment by davidanthoff** [comment]
Alright, I'm going to update the README and then release this to the marketplace, ok?

**Comment by davidanthoff** [comment]
@ZacLN Can you tag a CSTParser.jl version?

**Comment by ZacLN** [comment]
Yep will do
